detection_rules/rule.py

```diff
@@ -211,6 +234,19 @@ def ast(self):
         if validator is not None:
             return validator.ast
 
+    def get_beats_types(self) -> List[str]:
+        indexes = self.index or []
+        beat_types = [index.split("-")[0] for index in indexes if "beat-*" in index]
+        return beat_types
+
+    def get_query_schema(self, ast: Union[eql.ast.Expression, kql.ast.Expression], beats_version: str,
+                         ecs_version: str) -> dict:
+        indexes = self.index or []
+        beat_types = self.get_beats_types()
+        beat_schema = beats.get_schema_from_kql(ast, beat_types, version=beats_version) if beat_types else None
```

Comment:
not necssarily KQL

Reply:
can you move to the query validator instead? that checks the language more carefully, and the class hiearchy is a little flawed.

Reply:
7a90cc3